Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLParameterScan.py

```diff
@@ -0,0 +1,233 @@
+
+    def seeAlso(self):
+        return []
+
+    def name(self):
+        return 'SANSILLParameterScan'
+
+    def validateInputs(self):
+        issues = dict()
+        message = "Wrong number of {0} runs: {1}. Please provide none or exactly one."
+        sens_dim = self.getPropertyValue('SensitivityMap').count(',')
+        if not (self.getPropertyValue('OutputJoinedWorkspace') or self.getPropertyValue("OutputWorkspace")):
+            issues["OutputJoinedWorkspace"] = "Please provide either OutputJoinedWorkspace, OutputWorkspace or both."
+            issues["OutputWorkspace"] = "Please provide either OutputJoinedWorkspace, OutputWorkspace or both."
+        if sens_dim != 0:
+            issues['SensitivityMap'] = message.format("sensivity map", sens_dim)
+        if self.getPropertyValue('PixelYmin') > self.getPropertyValue("PixelYmax"):
+            issues["PixelYMin"] = "YMin needs to be lesser than YMax"
+            issues["PixelYMax"] = "YMax needs to be greater than YMin"
+        return issues
+
+    def setUp(self):
+        self.sample = self.getPropertyValue('SampleRuns')
+        self.absorber = self.getPropertyValue('AbsorberRuns').replace('+', ',')
+        self.container = self.getPropertyValue('ContainerRuns').replace('+', ',')
+        self.sensitivity = self.getPropertyValue('SensitivityMap')
+        self.default_mask = self.getPropertyValue('DefaultMaskFile')
+        self.normalise = self.getPropertyValue('NormaliseBy')
+        self.output2D = self.getPropertyValue('OutputWorkspace')
+        self.output_joined = self.getPropertyValue('OutputJoinedWorkspace')
+        self.observable = self.getPropertyValue('Observable')
+        self.pixel_y_min = self.getProperty('PixelYMin').value
+        self.pixel_y_max = self.getProperty('PixelYMax').value
+        self.progress = Progress(self, start=0.0, end=1.0, nreports=10)
+
+    def checkPixelY(self, height):
+        if self.pixel_y_max > height:
+            self.pixel_y_max = height
+            logger.warning("PixelYMax value is too high. Reduced to {0}.".format(self.pixel_y_max))
+
+    def PyInit(self):
+
+        self.declareProperty(WorkspaceProperty('OutputWorkspace', '', direction=Direction.Output,
+                                               optional=PropertyMode.Optional),
+                             doc="The output workspace containing the 2D reduced data.")
+
+        self.declareProperty(WorkspaceProperty('OutputJoinedWorkspace', '', direction=Direction.Output,
+                                               optional=PropertyMode.Optional),
+                             doc="The output workspace containing all the reduced data, before grouping.")
+
+        self.declareProperty(MultipleFileProperty('SampleRuns',
+                                                  action=FileAction.Load,
+                                                  extensions=['nxs'],
+                                                  allow_empty=False),
```

Comment:
You can remove the allow_empty, this is for different purpose; just `FileAction.Load` is enough to make it mandatory.